Redesign Part 5

willpill opened the pull request:
This PR, at the base level, tries to refactor the codebase to improve organization and clean up various things. Added on top are major UI updates with new components, fully functional theming, new features such as Activity Recommendations, and minor content updates.

## Codebase Organization & File Structure

Completely updated the structure throughout the codebase for consistency. Consolidated and renamed files and directories.

### Assets

`public` folder assets are now categorized into:

- `achievements` for medal displays
- `symbols`
- `icons`
- `taglines`
- etc.

Images have been renamed as well to follow convention.

### Components

Right now under `src/pages/modules` the code splits functionality into modules.  Each module contains its own router and pages but rely on `comps/pages`.  There is also a separate `pages/admin` for administrative tools.

This mixing of UI components and page specific components inside a single comps folder is confusing.  Even more so when such comps are mixed with comps stored in individual pages.

So, we adopt co‑location here, keeping each page’s components in one folder and separate shared UI, contexts, utilities, services and types.  

- We split generic vs feature components. Keep reusable UI controls, dialogs, navigation and other generics in `src/components`. It's further separated into each component, such as buttons, which will be helpful for our full design system implementation.

- `comps/pages` is no longer needed. We colocate specific components under the pages that use them. Under `src/modules`, we very clearly compartmentalize each sector of Epsilon. There is a directory for each top‑level route (about, home, stuyactivities, etc.).  The route component is in `index.tsx` and the page’s components are stored in a `components/` subfolder.  For example, the current Home.tsx becomes src/modules/home/index.tsx; its children `UnauthenticatedLanding`, etc. move into its own `components/` subfolder. Meanwhile `OrgBlock` is a shared, small component used across sectors, so it is moved to the generic comp folder.

- For stuyactivities, we add a new page for requesting unaffiliated room use with the intent to replace the ad-hoc activity. We split the module into:
> admin

>> index

>> components folder

>> pages folder

> components

> orgs

>> components folder

>> org_admin folder for org admin actions

>>> components

>>> pages

>> pages folder for general org pages like Charter info

> Surface Level Pages such as Catalog & Archives

- Providers & contexts are not UI. Moved to `src/contexts`. Services, etc. are moved to their own folders.


### Modules

- We make each sector of Epsilon its own module, which makes everything easier for the future. StuyActivities is now considered a module as well, along with the Home module, Calendar module, Opportunities module (future), Valentines module, etc.

- Attendance being its own module, completely separate from StuyActivities, does not make sense, so we consolidated it into each Org. In the Admin Tools section of each `OrgNav`, there is now an attendance option which shows  `AttendanceOverview`, it may trigger individual `MeetingAttedance` options which works as usual. Members will be able to take their own attendance as usual via `MyMeetingAttendance`.

## UI

### Reusable Components for #294
Created new reusable components: `ConfirmationDialog`, `ContentUnavailable`, `ContextualNavBar`, `Divider`, `ItemList`, etc. that are used across Epsilon.

### Catalog
- Redesigned OrgCards to include more info like commitment, and also adjusted size and title displays to ensure 3-line title activities still look good.
- OrgCards now load in with a subtle blur + shift animation
- Fixed the old issue where upon loading more Activities, the user is snapped back up the page. Now scroll position is kept.

### User
- New Preferences Page
- Redesigned User Page (Epsilon Passport) with brand new medals
- Redesigned Communications Page

### Theming, enough that we can consider it closes #293
- Moved all design system colors and styles into Tailwind
- Created 3 Themes, Light, Dark, High Contrast, and there is also an Auto mode (system). These are available to switch between in the new User Preferences page.
- Parameterized most colors.

<img width="725" height="316" alt="Screenshot 2025-08-22 at 4 21 35 PM" src="https://github.com/user-attachments/assets/72f38bca-648d-4886-80bb-b622a1268521" />


### Migration
- Important pages mostly migrated out of MUI.
- `ThemeProvider` repurposed, while still supporting lingering MUI usage minimally.
- Various components migrated out of MUI. New components are based on Radix Primitives as per #295 and designed with smooth animations and more reusability, such as Org Popups.

### Responsive Behavior
- Greatly improved responsive behaviors across multiple pages.
- Activity pages now show the `OrgInspector` across the right side on xl and above breakpoints only. Otherwise, it is included in Overview & Audit pages, respectively.
- Activity pages now have better treatments on smaller breakpoints, with limited widths for reading comfort.

### Other
- Skeleton Loading Implemented for OrgCard and OrgBlock
- Improved `NavBar` user display, animates dropdown icon.
- Adjusted text sizes
- Fixes an issue that causes animations to play before images are loaded
- Fixes to spacings and highlights across UI elements so that m-12 is standard for the main content.
- Removed various unused or redundant UI components
- Other UI updates

## Recommendations to fix #95 

- A new `OrgListProvider` context is created and we wrap `Pages` with it in for storing the list of activities used for recommendations.
- We first strip out filler words like "stuyvesant", "club", etc.
- Name match up to 4.5 pts
- Purpose similarity up to 3.5 points
- Percentage overlap of tags * 1.5 points
- Commitment level match + 0.5 points
- Normalized from 10 to 100 pts, with a small curve, but also a cap, with the top matches shown.

## Auto Refresh on the following to fix #219 
- Messages
- Attendance

## Naming & Copy
- Updated navigation routes and labels.
- Updated various UI texts
- Fixed various grammatical errors
- Added help sections

## SEO
- Added various title and description tags
- Dynamic title and description tags for Activity pages

willpill commented on public/index.html:
removing to allow for page specific meta tags

willpill commented on src/components/ui/orgs/OrgBlock.tsx:
replaces previous orgbar, used by recs and homepage

willpill commented on src/components/ui/orgs/MeetingPreview.tsx:
meetingpreview dialog w radix

willpill commented on src/components/ui/ItemList.tsx:
abstraction of item lists to this

willpill commented on src/components/ui/Loading.tsx:
use lottie for loading animation

willpill commented on src/config/routes.ts:
move routes to separate file

willpill commented on src/contexts/OrgListContext.tsx:
store context of all orgs to allow for quick recs

willpill commented on src/modules/about/index.tsx:
completely redesign for animated credits

willpill commented on src/modules/arista/index.tsx:
arista landing page only

willpill commented on src/components/ui/ToggleChip.tsx:
custom chip implementation to replace MUI

willpill commented on src/components/ui/input/ToggleChip.tsx:
Component for chip

willpill commented on src/components/ui/lists/ItemList.tsx:
Component for Lists, along with OverviewList

willpill commented on src/modules/stuyactivities/admin/index.tsx:
Removed MUI icons altogether

willpill commented on src/styles/app.css:
global styles

willpill reviewed (commented):
Significant changes are noted

SnowyNate commented:
Haven't reviewed this for bugtests yet, but received some feedback from students from the CURRENT version of Epsilon UI, not this repo and just wanted to list them here for further consideration:

- Remove blur when hovering over main menu items
- Have links open in new tab, rather than in current Epsilon window, add "open in new tab" icon to links (e.g. Opportunities)
- Remove points system

SnowyNate commented:
Beautiful, thank you

I understand the purposes of the menu hover blur, but it is quite distracting and I think having the button glow when hovering is enough- would it be possible to add a temporary commit so I can see how things look?

SnowyNate reviewed (changes requested):
Overall, great work! Thank you very much for the awesome changes, and the new themes are beautiful, especially light mode. Tested successfully, just a couple of issues/improvements listed below:  

**StuyActivities > Admin Panel**
- Scrollbars appearing on menu icons
<img width="2559" height="516" alt="Screenshot 2025-08-29 225904" src="https://github.com/user-attachments/assets/48432bb0-447c-4a74-a1b6-79a278b93715" />

- Expand menu icons to show what they mean please, just icons are confusing  
- Improve Rooms UI layout

**About**
- Speed up animation
- Looks strange in light mode (improve gradients?)

**StuyActivities > Submenu items**
- Is there any way to make the submenu items clearer? (e.g. add underlining lines under the items such as in Epsilon main menu options)

**Main Menu**
- Reword menu items (Apply --> Apply for SU)
- When navigating away, then returning to Epsilon using the browser's back button, main menu bar breaks and is misaligned
<img width="2559" height="358" alt="image" src="https://github.com/user-attachments/assets/bdc3d971-9ce0-440f-9253-6b41fcc59c0e" />

**Profile Dropdown**
- Reword "My Epsilon Passport" back to "Profile" and change "My Epsilon Passport" and "Passport ID" back to just "Profile" and "ID"
- Still unsure of the necessity of Epsilon Medals...

**StuyActivities > Club > Club options**
- Club options overflows into footer when scrolling
<img width="2557" height="1316" alt="image" src="https://github.com/user-attachments/assets/7d343182-855a-46d0-9adc-235f74657d50" />

- Scrollbars around club statistics (happens upon viewing a club as well)
<img width="1513" height="279" alt="image" src="https://github.com/user-attachments/assets/1487d8ec-2fdc-4b3e-a7eb-484cab770ed9" />

- Scheduler and Attendance look strangely dark and don't follow the same format as other menu options in light mode and custom themes
<img width="2552" height="1344" alt="image" src="https://github.com/user-attachments/assets/7eeff3bc-2ade-49fc-b7a3-4daddfea902c" />

**Communications**
- Allow disabling of Epsilon Notices or don't list as an option

willpill commented:
I've addressed all of these issues, except for submenu items, which I think should remain the same unless some complaint arises about it being confusing

SnowyNate commented:
THANK YOU SO MUCH FOR THE EXCELLENT WORK!!!! In regards to the submenu items, yes I have received some complaints already hence why I suggested adding some sort of divider or highlighted line underneath them if possible

Just a couple more issues before we finally push to main, but the previous issues were addressed very well! (love the loading animation btw)

**StuyActivities > Admin Panel**
- [ ] Footer bug where footer appears too early on some admin panel menu items _(Pending Orgs, Approve Edits, Announcements, Rooms, Reserve)_, leaving too much whitespace at the bottom
<img width="1920" height="959" alt="image" src="https://github.com/user-attachments/assets/b7d1af1f-9f07-4afa-ac18-b4dc27a839e4" />

- [ ] Scrollbar issue with Rooms
<img width="1869" height="244" alt="image" src="https://github.com/user-attachments/assets/b2092cb0-394a-41a7-9404-053375f3512e" />

**Home**
- [ ] Home page looks squished and doesn't adapt well to larger screens (smaller screens seem to be ok), items on page look misordered and squished
<img width="1920" height="957" alt="image" src="https://github.com/user-attachments/assets/27898e47-985e-4340-a16e-80d2f73818fb" />

- [ ] Scrollbars issue with My Meetings item when there are listed meetings

**Profile**
- [ ] Medals of Honor gradient is not visible on light mode
<img width="543" height="117" alt="image" src="https://github.com/user-attachments/assets/29803697-ea4d-4319-8c41-b58231328150" />

- [ ] Share Medal button doesn't really do anything, could you remove it temporarily until the feature is ready?

**Menu**
- [ ] Menu Items get cut off and there is no way to access them if viewing in a smaller window
<img width="743" height="734" alt="image" src="https://github.com/user-attachments/assets/87c5fccc-b954-4e02-871b-a55ffaf006f6" />

**Calendar**
- [ ] Scrollbars issue with listed meetings

**StuyActivities > Club > Club options**
- [ ] In Attendance, Make "Present" or "Absent" indicators color-coded (green for present and red for absent) for ease of use
- [ ] There's a sharp gradient change in Attendance between the status indicator and the button
<img width="322" height="98" alt="image" src="https://github.com/user-attachments/assets/5bfe58c7-e4c7-4bcd-9ab8-43be548bce96" />

- [ ] Add copy button for link in Attendance
- [ ] Join Requests request is a sharp contrast from the background
<img width="1389" height="281" alt="image" src="https://github.com/user-attachments/assets/ab6c370f-94ed-49c4-b8ac-48aa009f7c45" />

- [ ] Extra 0 digit for ID in Attendance
<img width="1357" height="173" alt="image" src="https://github.com/user-attachments/assets/ee48cd3e-fd28-492c-92a0-955b153459e1" />

- [ ] Buttons for a post are very squished, especially when the post is small
<img width="237" height="418" alt="image" src="https://github.com/user-attachments/assets/a4e34e5b-1d60-4414-b273-1d937d4bec55" />

- [ ] Scrollbars issue for large posts, lots of extra white space _(happens on home page too)_
<img width="1351" height="398" alt="image" src="https://github.com/user-attachments/assets/004a3bd3-6a9e-4a12-8c5e-76011c7fa699" />

- [ ] Scrollbars issue for club statistics (still persistent from last comment)
- [ ] Room Undefined issue for a Virtual meeting room
<img width="916" height="151" alt="image" src="https://github.com/user-attachments/assets/a24a8934-aa19-488a-bba2-ecaf7b7e1cc8" />

- [ ] Clicking on club name when viewing a meeting takes user to an undefined page

willpill commented:
I will be fixing these as soon as possible

willpill commented:
Fixed everything; scrollbar with rooms is intentional.
With other scrolling issues, I either added a side gradient to indicate scrollability, or disabled scrollbars from showing up.

SnowyNate commented:
Beautiful work, will take a look at these changes later and hopefully push to main if all looks well

willpill commented:
I removed the Navbar scroll gradient since it causes issues on some pages. While there might be better solutions to indicate scrollability, I won't be able to implement them for now.

SnowyNate commented:
How about using our custom scrollbar rather than native scrollbars so it at least looks more appealing?

SnowyNate commented:
Testing now, everything looks good enogh to push to main but I caught one major bug:

**Activity Meetings Scheduler**

- [ ] When trying to update a meeting, the meeting description pops up at the same time the edit menu appears, which is unclickable until the user clicks elsewhere and the meeting description goes away
